src/main/java/DukePackage/Storage.java

```diff
@@ -0,0 +1,224 @@
+                        break;
+                    case DEADLINE:
+                        formattedString = String.format("%c|%d|%s|%s",
+                                'D', priority, tasking.description,
+                                tasking.startTime.toString().replace("T", " "));
+                        break;
+                    case EVENT:
+                        formattedString = String.format("%c|%d|%s|%s|%s",
+                                'E', priority, tasking.description,
+                                tasking.startTime.toString().replace("T", " "),
+                                tasking.endTime.toString().replace("T", " "));
+                        break;
+                }
+                bufferedWriter.write(formattedString);
+                bufferedWriter.newLine(); // Move to the next line
+            }
+            bufferedWriter.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Loads the task list from a file.
+     */
+    public void loadListFromFile() {
+        try (BufferedReader reader = new BufferedReader(new FileReader("data/duke.txt"))) {
+            File file = new File("data/duke.txt");
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            FileReader fileReader = new FileReader(file); // append mode
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String line;
+            while ((line = reader.readLine()) != null) {
+                // Assuming your line contains comma-separated values
+                String[] values = line.split("\\|");
+                // Create your Java object based on the parsed values
+                TaskType type = Objects.equals(values[0], "T")
+                        ? TaskType.TODO
+                        : Objects.equals(values[0], "D")
+                        ? TaskType.DEADLINE
+                        : TaskType.EVENT;
+                String start = "", end = "";
+                try {
+                    start = values[3];
+                } catch (Exception e) {
+                    start = "";
+                }
+                try {
+                    end = values[4];
+                } catch (Exception e) {
+                    end = "";
+                }
+                Task obj = new Task(values[2], type, start, end); // Instantiate with appropriate arguments
+                obj.setStatus(!Objects.equals(values[1], "0"));
+                // Store the object in your storage instance
+                addList(obj);
+                bufferedReader.close();
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Prints the task list.
+     */
+    public void printTaskList() {
+        for (int i = 0; i < this.taskList.size(); i++) {
+            int index = i + 1;
```

Comment:
Think it would be great if you could add a comment here explaining the rationale for + 1! 😺